missing-campaign.js

```diff
@@ -0,0 +1,205 @@
+                ]
+            }
+        }
+    }
+
+    const firstValidationRule = {
+        "id": null,
+        "name": "Business Validation Rule - Reward Promotion Tier 1",
+        "rules": {
+            "1": {
+                "name": "order.amount",
+                "property": null,
+                "conditions": {
+                    "$more_than": [
+                        10000
+                    ]
+                }
+            },
+            "2": {
+                "name": "order.amount",
+                "property": null,
+                "conditions": {
+                    "$is": [
+                        10000
+                    ]
+                }
+            },
+            "logic": "(1 or 2)"
+        },
+        "error": null,
+        "applicable_to": {
+            "included": [],
+            "excluded": [],
+            "included_all": false
+        },
+        "type": "advanced",
+        "context_type": "campaign.promotion"
+    }
+
+    const secondValidationRule = {
+        "id": null,
+        "name": "Business Validation Rule - Reward Promotion Tier 2",
+        "rules": {
+            "1": {
+                "name": "order.amount",
+                "property": null,
+                "conditions": {
+                    "$more_than": [
+                        25000
+                    ]
+                }
+            },
+            "2": {
+                "name": "order.amount",
+                "property": null,
+                "conditions": {
+                    "$is": [
+                        25000
+                    ]
+                }
+            },
+            "logic": "(1 or 2)"
+        },
+        "error": null,
+        "applicable_to": {
+            "included": [],
+            "excluded": [],
+            "included_all": false
+        },
+        "type": "advanced",
+        "context_type": "campaign.promotion.discount.apply_to_order"
+    }
+
+    const thirdValidationRule = {
+        "id": null,
+        "name": "Business Validation Rule - Reward Promotion Tier 3",
+        "rules": {
+            "1": {
+                "name": "order.amount",
+                "property": null,
+                "conditions": {
+                    "$more_than": [
+                        50000
+                    ]
+                }
+            },
+            "2": {
+                "name": "order.amount",
+                "property": null,
+                "conditions": {
+                    "$is": [
+                        50000
+                    ]
+                }
+            },
+            "logic": "(1 or 2)"
+        },
+        "error": null,
+        "applicable_to": {
+            "included": [],
+            "excluded": [],
+            "included_all": false
+        },
+        "type": "advanced",
+        "context_type": "campaign.promotion.discount.apply_to_order"
+    }
+
+    const createRewardPromotion = async () => {
+        await Promise.all([
```

Comment:
You mixed two approaches:

chaining promises
```ts
const result1 = await somePromise1();
const result2 = await somePromise2();
const result3 = await somePromise3();
```

parallel promises
```ts

const results = await Promise.all([
  somePromise1(),
  somePromise2(),
  somePromise3()
])
```


You also mixed `async/await` syntax with `then` syntax.

And you miss promise handling for `client.campaigns.create`

At the end:


```ts

const validationRules = await Promise.all([
  client.validationRules.create(firstValidationRule),
  client.validationRules.create(secondValidationRule),
  client.validationRules.create(thirdValidationRule),
])

const rewardPromotion = createRewardPromotionObject(validationRules[0].id, validationRules[1].id, validationRules[2].id);

const response = await client.campaigns.create(rewardPromotion);

if (response.code !== 200) {
  return new Error;
}

```

Reply:
changed, but have to check one thing